### PR TITLE
DSD-1582: better association between the SearchBar's helper text and input element

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1582](https://jira.nypl.org/browse/DSD-1582)

## This PR does the following:

- Updates the `TextInput` and `SearchBar` components to better associate the input element to the entire component's helper text. This is done by overriding the `TextInput`'s internally generated `aria-describedby` value when there is an explicit value added through the `aria-describedby` prop. 

## How has this been tested?

Storybook

## Accessibility concerns or updates

- This was flagged as "Descriptive text is not programmatically associated (major)"

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -270,15 +272,20 @@ export const TextInput: ChakraComponent<
...
-      });
+      const ariaAttributes = {
+        ...getAriaAttrs({
+          additionalAriaLabel,
+          footnote,
+          id,
+          labelText,
+          name: "TextInput",
+          additionalHelperTextIds,
+          showLabel,
```
Comment (bigfishdesign13) [inline src/components/TextInput/TextInput.tsx]: Nit: Is there a reason these are not alphabetized?

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.  And approved by Clare!

File: src/components/TextInput/TextInput.tsx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/TextInput/TextInput.tsx]: No, can update.

**Comment by EdwinGuzman** [comment]
Bump for review @jackiequach @7emansell @oliviawongnyc

**Review by jackiequach — APPROVED** [review]
Looks good!

**Pull request merged**